packages/terra-application-layout/src/ApplicationLayout.jsx

```diff
@@ -0,0 +1,236 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import AppDelegate from 'terra-app-delegate';
+import NavigationLayout from 'terra-navigation-layout';
+import { routeConfigPropType } from 'terra-navigation-layout/lib/configurationPropTypes';
+import { matchPath } from 'react-router-dom';
+import { withModalManager } from 'terra-modal-manager';
+
+import RoutingMenu from './menu/RoutingMenu';
+import ApplicationMenuWrapper from './menu/_ApplicationMenuWrapper';
+import ApplicationHeader from './header/_ApplicationHeader';
+import ApplicationLayoutPropTypes from './utils/propTypes';
+import Helpers from './utils/helpers';
+import UtilityHelpers from './utils/utilityHelpers';
+
+const navigationLayoutSizes = ['default', 'tiny', 'small', 'medium', 'large', 'huge'];
+
+const propTypes = {
+  /**
+   * The AppDelegate instance provided by `withModalManager`. If an AppDelegate instance is
+   * explicitly provided to the ApplicationLayout, the ModalManager will wrap it and
+   * fall back to its defined APIs as needed.
+   */
+  app: AppDelegate.propType,
+  /**
+   * The content to be rendered in the ApplicationLayout's extensions region. This component will be provided an AppDelegate (as `app`) and
+   * a `layoutConfig` as props to facilitate communication with the ApplicationLayout.
+   */
+  extensions: PropTypes.element,
```

Comment:
Using `PropTypes.element` here prevents passing this a functional component. If you try and use [the example for the extensions prop](https://github.com/cerner/terra-framework/blob/application-layout-pr/packages/terra-application-layout/docs/usage.md#extensions) which uses the functional component, `MyExtensions`, it triggers a prop warning. 

```
Warning: Failed prop type: Invalid prop `extensions` of type `function` supplied to `ApplicationLayout`, expected a single ReactElement.
```